homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -145,21 +174,112 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
                         for variable in sensor_variables
                     ]
                 )
-
-    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+    elif config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
         host = config_entry.data[CONF_HOST]
         token = config_entry.data[CONF_TOKEN]
-        name = config_entry.title
-        unique_id = config_entry.unique_id
-
-        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+        model = config_entry.data[CONF_MODEL]
+        device = None
+        if model in MODELS_HUMIDIFIER_MIOT:
+            device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+            coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
```

Comment:
The coordinator is the same for all these cases.

Reply:
Yes, but `hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]` is not initialized for the AirQualityMonitor.